storage/innobase/dict/dict0stats.cc

```diff
@@ -1754,6 +1835,8 @@ dict_stats_scan_page(
 	Because offsets1,offsets2 should be big enough,
 	this memory heap should never be used. */
 	mem_heap_t*	heap			= NULL;
+	const ulint	n_core = leaf ? index->n_core_fields : 0;
+	ut_ad(leaf_stats);
```

Comment:
Can we declare the function as `__attribute__((nonnull))` and omit this debug assertion?